modules/migrate/application/clicommands/ToicingadbCommand.php

```diff
@@ -0,0 +1,596 @@
+        $hostActions = $this->readFromIni($sharedNavigation . '/host-actions.ini', $rc);
+        $serviceActions = $this->readFromIni($sharedNavigation . '/service-actions.ini', $rc);
+        $icingadbHostActions = $this->readFromIni($sharedNavigation . '/icingadb-host-actions.ini', $rc);
+        $icingadbServiceActions = $this->readFromIni($sharedNavigation . '/icingadb-service-actions.ini', $rc);
+
+        Logger::info('Transforming legacy wildcard filters of existing shared Icinga DB Web actions');
+
+        if (! $icingadbHostActions->isEmpty()) {
+            $this->migrateNavigationItems($icingadbHostActions, true, $rc);
+        }
+
+        if (! $icingadbServiceActions->isEmpty()) {
+            $this->migrateNavigationItems(
+                $icingadbServiceActions,
+                true,
+                $rc
+            );
+        }
+
+        Logger::info('Migrating shared monitoring navigation items to the Icinga DB Web actions');
+
+        if (! $hostActions->isEmpty()) {
+            $this->migrateNavigationItems($hostActions, true, $rc, $sharedNavigation . '/icingadb-host-actions.ini');
+        }
+
+        if (! $serviceActions->isEmpty()) {
+            $this->migrateNavigationItems(
+                $serviceActions,
+                true,
+                $rc,
+                $sharedNavigation . '/icingadb-service-actions.ini'
+            );
+        }
+
+        if ($rc > 0) {
+            Logger::error('Failed to migrate some monitoring navigation items');
+            exit($rc);
+        }
+
+        Logger::info('Successfully migrated all local user monitoring navigation items');
+    }
+
+
+    /**
+     * Migrate monitoring restrictions in a role to Icinga DB Web restrictions
+     *
+     * USAGE
+     *
+     *  icingacli migrate toicingadb restriction [options]
+     *
+     * OPTIONS:
+     *
+     *  --group=<groupname>  Migrate monitoring restrictions for all roles,
+     *                       the given group or the groups matching the given
+     *                       group belongs to.
+     *                       (wildcard * migrates monitoring restrictions
+     *                       for all roles)
+     *
+     *  --role=<rolename>    Migrate monitoring restrictions for the
+     *                       given role or all the matching roles.
+     *                       (wildcard * migrates monitoring restrictions
+     *                       for all roles)
+     *
+     *  --override          Override the existing Icinga DB restrictions
+     */
+    public function restrictionAction(): void
+    {
+        $override = $this->params->get('override');
+
+        /** @var ?string $groupName */
+        $groupName = $this->params->get('group');
+        /** @var ?string $roleName */
+        $roleName = $this->params->get('role');
+
+        if ($roleName === null && $groupName === null) {
+            $this->fail("One of the parameters 'group' or 'role' must be supplied");
```

Comment:
please also fail if both are given, since --role overrules --group, this is better than just ignoring --group